### PR TITLE
Add poetry venv files

### PR DESCRIPTION
**Pull request by nick-mobilecoin**
### Motivation

Add poetry for use as a virtual environment manager. This will provide a
consistent environment for developers.

### Future Work

- Finish the rest of #25

### CODE REVIEW

**Comment by nick-mobilecoin** [comment]
Current dependencies on/for this PR:
* main
  * **PR #27** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/cookiecutters/27" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/mobilecoinfoundation/cookiecutters/27?utm_source=stack-comment).

File: .github/dependabot.yaml [hunk 1]
```diff
@@ -13,3 +13,9 @@ updates:
       interval: daily
     commit-message:
       prefix: "chore(deps)"
+  - package-ecosystem: pip
```
Comment (nick-mobilecoin) [inline .github/dependabot.yaml:16]: not sure if we want dependabot updating all of the python stuff, but don't think it should hurt.
If we need to support older versions then we can make a CI run dedicated to the older one.

File: pyproject.toml [hunk 2]
```diff
@@ -0,0 +1,17 @@
+[tool.poetry]
+name = "cookiecutters"
+version = "0.1.0"
+description = "MobileCoin cookiecutter templates"
+authors = ["MobileCoin"]
+license = "Apache-2.0"
```
Comment (nick-mobilecoin) [inline pyproject.toml:6]: Poetry assumes a package will be published, I don't think we'll be doing that, but left this here anyway

File: .github/dependabot.yaml (hunk 1 above)
Comment (jcape) [reply .github/dependabot.yaml:16]: Would this conflict with `poetry.lock`?

Comment (nick-mobilecoin) [reply .github/dependabot.yaml:16]: no, dependabot uses 'pip' for *all* python deps, https://docs.github.com/en/code-security/dependabot/dependabot-version-updates/configuration-options-for-the-dependabot.yml-file#package-ecosystem
image

**Review by jcape — APPROVED** [review]

**Pull request merged**